circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -216,7 +227,7 @@ private class CircuitSymbolProcessor(
         className = name
         packageName = fd.packageName.asString()
         factoryType =
-          if (name.endsWith("Presenter")) {
```

Comment:
I realize that the existing solution expects the state producing function to have "Presenter" in the name. Instead of doubling down on that approach I think a better solution would be to verify that the function returns a single object extending `CircuitUiState`. This will allow us to identify the presenter as the UI will have no return value.

Bonus: such an approach also wouldn't impose a naming convention on Circuit users

Reply:
What's the argument for not imposing a naming convention?  It definitely helps with readability.  Why are we imposing the "Presenter" naming convention in the first place?  If we are going to drop naming conventions that would strengthen the case for two different annotations IMO.